ksql-serde/src/test/java/io/confluent/ksql/serde/avro/KsqlGenericRowAvroSerializerTest.java

```diff
@@ -329,6 +341,53 @@ public void shouldSerializeMap() {
         avroValue);
   }
 
+  @Test
+  public void shouldSerializeMultipleMaps() {
+    final org.apache.avro.Schema avroInnerSchema0
+        = mapEntrySchema("KsqlDataSourceSchema_field0_inner0");
+    final org.apache.avro.Schema avroInnerSchema1 =
+        mapEntrySchema(
+            "KsqlDataSourceSchema_field0_inner1",
+            org.apache.avro.SchemaBuilder.unionOf().nullType().and().stringType().endUnion());
+    final org.apache.avro.Schema avroSchema =
+        org.apache.avro.SchemaBuilder.record("KsqlDataSourceSchema_field0")
+            .namespace("io.confluent.ksql.avro_schemas")
+            .fields()
+            .name("inner0").type().unionOf().nullType().and().array()
```

Comment:
I'm a little confused - this makes me think that the schema is an array of maps, but then below when we create the KSQL schema it seems to be just a single map. Am I missing something?

Reply:
Avro only lets you represent maps with string keys natively. So to work around this, connect will represent maps as an array of key-value pairs.

Here's what the avro schema for this test looks like:
```
{
	"type": "record",
	"name": "KsqlDataSourceSchema_field0",
	"namespace": "io.confluent.ksql.avro_schemas",
	"fields": [{
		"name": "inner0",
		"type": ["null", {
			"type": "array",
			"items": {
				"type": "record",
				"name": "KsqlDataSourceSchema_field0_inner0",
				"fields": [{
					"name": "key",
					"type": ["null", "string"],
					"default": null
				}, {
					"name": "value",
					"type": ["null", "long"],
					"default": null
				}],
				"connect.named.map": "true"
			}
		}],
		"default": null
	}, {
		"name": "inner1",
		"type": ["null", {
			"type": "array",
			"items": {
				"type": "record",
				"name": "KsqlDataSourceSchema_field0_inner1",
				"fields": [{
					"name": "key",
					"type": ["null", "string"],
					"default": null
				}, {
					"name": "value",
					"type": ["null", "long"],
					"default": null
				}],
				"connect.named.map": "true"
			}
		}],
		"default": null
	}]
}
```

Reply:
Thanks @rodesai - this helps a lot.